src/main/resources/archetype-resources/__rootArtifactId__/src/main/java/__pluginFormat__/internal/proxy/__pluginClass__ProxyFacetImpl.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) ${currentYear}-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package ${package}.${pluginFormat}.internal.proxy;
+
+import java.io.IOException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.sonatype.nexus.plugins.${pluginFormat}.internal.AssetKind;
+import org.sonatype.nexus.plugins.${pluginFormat}.internal.util.${pluginClass}DataAccess;
+import org.sonatype.nexus.plugins.${pluginFormat}.internal.util.${pluginClass}PathUtils;
+import org.sonatype.nexus.repository.cache.CacheInfo;
+import org.sonatype.nexus.repository.config.Configuration;
+import org.sonatype.nexus.repository.proxy.ProxyFacet;
+import org.sonatype.nexus.repository.proxy.ProxyFacetSupport;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.Bucket;
+import org.sonatype.nexus.repository.storage.Component;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.nexus.repository.storage.TempBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalStoreBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalTouchBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalTouchMetadata;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.matchers.token.TokenMatcher;
+import org.sonatype.nexus.transaction.UnitOfWork;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.plugins.${pluginFormat}.internal.util.${pluginClass}PathUtils.*;
```

Comment:
Trivial: We dont normally include `*`

Reply:
good catch! This is sort of a side-effect of my trying to come up with a good "default" stub, so not really sure which things to import yet - because not really sure which things to define in PathUtils...and round and around I go. ;)

Reply:
fixed here: f427df0